### PR TITLE
Enable securityContext for pod and container

### PR DESCRIPTION
**Pull request by rodrigobrim**
As part of our plan to run the container as an ordinary user, we need to change the default `container[0].securityContext.privileged` from true to false.

We also need the ability to change the pod security context, allowing the `fsGroup` storage parameter.

### CODE REVIEW

**Review by jonyoder — APPROVED** [review]
Looks good to me. It would probably be wise to have someone more familiar with helm (and more involved with this repo) to also take a look.

**Comment by rodrigobrim** [comment]
> Looks good to me. It would probably be wise to have someone more familiar with helm (and more involved with this repo) to also take a look.

Sure. Aaron and Cole added.

**Review by colearendt — CHANGES_REQUESTED** [review]
Looks good! I didn't realize we still run this as privileged. A couple of things:

- it looks like you mixed up the convention with the other products. You have `pod.securityContext` setting the context on the container, and `securityContext` setting the context on the pod? I will need to take a deeper look at the other charts, but I think this is mixed up here (even the docs?)

- Also, if we are keeping the default as `privileged: true`, we should put that in the default values, rather than set the default in an `else` in the helm chart

**Comment by rodrigobrim** [comment]
> Looks good! I didn't realize we still run this as privileged. A couple of things:
> 
> * it looks like you mixed up the convention with the other products. You have `pod.securityContext` setting the context on the container, and `securityContext` setting the context on the pod? I will need to take a deeper look at the other charts, but I think this is mixed up here (even the docs?)
> * Also, if we are keeping the default as `privileged: true`, we should put that in the default values, rather than set the default in an `else` in the helm chart

They are not totally mixed. In the pod.securityContext there's no privileged config. In the container.securityContext, there's no fsGroup config. They are independent in some aspects.

About the if, I'll set it in the values.

**Comment by colearendt** [comment]
Oh no, my point is that you are using `pod.securityContext` as "container security context" and `securityContext` as "pod security context." Those should be reversed 😄 

Expected:
- `securityContext`: container security context
- `pod.securityContext`: pod security context

File: charts/rstudio-pm/values.yaml [hunk 1]
```diff
@@ -88,6 +88,9 @@ rstudioPMKey: false
 command: false
 # -- args is the pod's run arguments. By default, it uses the container's default
 args: false
+# -- container [securityContext](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) defines privilege and access control settings
+securityContext:
```
Comment (atheriel) [inline charts/rstudio-pm/values.yaml:92]: I've looked at [various](https://github.com/kubernetes/ingress-nginx/blob/24b8e32b088bec15e1f40deab8fefc8bd1b6c2da/charts/ingress-nginx/values.yaml#L825) [community](https://github.com/grafana/helm-charts/blob/f942ba605f8780518b6ac7f52eb68fb275c02e8b/charts/grafana/values.yaml#L101) charts and the convention is to use `containerSecurityContext` for this to disambiguate. Let's follow that convention.

**Review by atheriel — CHANGES_REQUESTED** [review]

**Review by atheriel — APPROVED** [review]
This looks fine to me now. @colearendt do you have an opinion on whether `containerSecurityContext` should be top-level or nested under `pod`? Are these changes in line with what you expected?

**Review by colearendt — APPROVED** [review]
I do like that convention within `pod`, I think, and consistency w/ the community is a win! We should take a follow-up issue to move our other charts towards this pattern.

**Pull request merged**